pkg/innerring/listen.go

```diff
@@ -0,0 +1,104 @@
+package innerring
+
+import (
+	"go.uber.org/zap"
+)
+
+// listenBlockchain runs asynchronous process which listens to the NeoFS-related
+// events thrown by the NeoFS blockchain network and schedules them for handling
+// by the local Inner Ring node. The background process is aborted by external
+// signal read from the provided channel.
+func (x *node) listenBlockchain(chDone <-chan struct{}) {
+	go x.listenNotifications(chDone)
+	go x.listenNotaryRequests(chDone)
+}
+
+// listenNotaryRequests listens to notary requests received from Notary service
+// of the NeoFS blockchain network, distills them to notaryRequest type
+// instances and schedules them for handling by the local Inner Ring
+// node according to related NeoFS smart contracts.
+//
+// The listenNotaryRequests is blocking: it's aborted by breaking the flow of
+// requests from the blockchain or by external signal read from the provided
+// channel.
+func (x *node) listenNotaryRequests(chDone <-chan struct{}) {
```

Comment:
As well as this one. Using regular subscription APIs you get exactly what you've subscribed to in the channel, almost as before (but #2254).